community/kernel/src/main/java/org/neo4j/kernel/impl/transaction/log/entry/LogEntryWriter.java

```diff
@@ -37,17 +37,34 @@
 public class LogEntryWriter
 {
     private final FlushableChannel channel;
+    private final LogEntryVersion version;
     private final Visitor<StorageCommand,IOException> serializer;
 
+    /**
+     * Create a writer that uses {@link LogEntryVersion#CURRENT} for versioning.
+     * @param channel underlying channel
+     */
     public LogEntryWriter( FlushableChannel channel )
     {
+        this( channel, CURRENT );
+    }
+
+    /**
+     * Create a writer that uses a different version than {@link LogEntryVersion#CURRENT}. Useful when writing test for
+     * migration/upgrade scenarios.
+     * @param channel underlying channel
+     * @param version version to put in the header
+     */
+    public LogEntryWriter( FlushableChannel channel, LogEntryVersion version )
```

Comment:
Hmm, this constructor is a bit dangerous if used in main (by mistake or otherwise). Could it be avoided somehow?

Reply:
Scary constructor indeed, I'll inline the needed channel writes instead.